test/runtests.jl

```diff
@@ -11,8 +11,36 @@ using FilePathsBase.TestPaths
 using UUIDs: uuid4
 using JSON3
 
-aws = AWSConfig()
+if VERSION >= v"1.5"
+    using Minio
+    AWS.aws_account_number(::Minio.MinioConfig) = "123"
 
+    # We run most tests under Minio. This can be done locally by those
+    # without access to the s3 bucket under which CI is performed.
+    # We then run all tests with s3 directly.
+
+    # We use multiple directories so that Minio can support versioning.
+    dirs = [mktempdir() for _ =1:12]
+    port = 9005
+    minio_server = Minio.Server(dirs; address="localhost:$port")
+
+    # We use this boolean flag to skip some tests under Minio
+    minio = true
+    try
+        run(minio_server, wait=false)
+        global aws = global_aws_config(MinioConfig("http://localhost:$port"; username="minioadmin", password="minioadmin"))
+        @testset "Minio tests" begin
+            include("s3path.jl")
+            include("awss3.jl")
+        end
+    finally
+        # Make sure we kill the server even if a test failed.
+        kill(minio_server)
+    end
+end
+
+aws = global_aws_config()
```

Comment:
My bad. This will just read the current config and not overwrite the `MinioConfig` when set:
```suggestion
# Use the `AWSConfig` for the following tests
aws = global_aws_config(AWSConfig())
```

Reply:
`config = ` or `cfg = ` may be a better more general name. I understand not wanting to change it though

Reply:
ah right. Yeah, it's used extensively in the tests. We could find and replace it though.